app/models/report.rb

```diff
@@ -14,4 +20,47 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def save_with_mentions
+    transaction do
+      save!
+      update_mentions!
+
+      true
+    end
+  rescue ActiveRecord::RecordInvalid
+    false
+  end
+
+  def update_with_mentions(params)
+    transaction do
+      update!(params)
+      update_mentions!
+
+      true
+    end
+  rescue ActiveRecord::RecordInvalid
+    false
+  end
+
+  private
+
+  def update_mentions!
+    mentioning_ids = content.scan(%r{http://localhost:3000/reports/(\d+)}).flatten.map(&:to_i).uniq
+
+    already_mentioning_ids = mentioning_reports.map(&:id)
+    new_mentioning_ids = mentioning_ids - already_mentioning_ids
+    deleted_mentioning_ids = already_mentioning_ids - mentioning_ids
+
+    new_mentioning_ids.each do |mentioning_id|
+      new_mentioning_report = Report.find_by(id: mentioning_id)
+      mentioning_reports << new_mentioning_report unless new_mentioning_report.nil?
```

Comment:
```suggestion
      mentioning_reports << new_mentioning_report if new_mentioning_report
```

または

```suggestion
      mentioning_reports << new_mentioning_report if new_mentioning_report.present?
```

のようになるべくunlessよりもifを使った方が読みやすいと思います